huntsman/camera/pyro.py

```diff
@@ -335,30 +310,22 @@ def _readout(self, filename=None):
         """Dummy method on the client required to overwrite @abstractmethod"""
         pass
 
-    def _async_wait(self, future_result, name='?', event=None, timeout=None):
-        # For now not checking for any problems, just wait for everything to return (or timeout)
-        if future_result.wait(timeout):
-            try:
-                result = future_result.value
-            except Exception as err:
-                # Add some extra text to the exception then re-raise it.
-                if len(err.args) >= 1:
-                    msg = f"Problem while waiting for {name} on {self.port}: {err.args[0]}"
-                    err.args = (msg,) + err.args[1:]
-                else:
-                    msg = f"Problem while waiting for {name} on {self.port}: {err}"
-                self.logger.error(msg)  # Make sure error makes it into the logs
-                raise err
-            finally:
-                if event is not None:
-                    event.set()
-        else:
-            if event is not None:
-                event.set()
-            msg = "Timeout while waiting for {} on {}".format(name, self.port)
-            raise error.Timeout(msg)
+    def _timeout_response(self, timeout_type):
+        # This could do more thorough checks for success, e.g. check is_exposing property,
+        # check for existence of output file, etc. It's supposed to be a last resort though,
+        # and most problems should be caught elsewhere.
+        relevant_event = getattr(self, f"_{timeout_type}_event")
+        is_set = True
+        try:
+            is_set = relevant_event.is_set()
+        except Pyro4.errors.CommunicationError:
+            # Can happen is everything has finished and shutdown before timeout,
+            # e.g. when running tests.
+            pass
```

Comment:
```suggestion
        # Exception can happen if everything has finished and shutdown before 
        # timeout, e.g. when running tests.
        with suppress(Pyro4.errors.CommunicationError):
            is_set = relevant_event.is_set()
```

Reply:
Done, but with some other overlapping changes so not using this suggested commit.